lib/models/service.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class Service < ActiveRecord::Base
+  has_many :exceptions, class_name: 'ServiceException'
+  has_many :trips
+
+  validates :hastus_id, presence: true, uniqueness: true
+  validates :start_date, :end_date, presence: true
+
+  serialize :weekdays, Array
+  validate :weekdays_format
+
+  def self.added_on(date)
+    joins(:exceptions).where(service_exceptions: { date: date,
+                                                   exception_type: 'add' })
+  end
+
+  def self.import(records)
+    records.each do |data|
+      where(data).first_or_create!
+    end
+  end
+
+  def self.on(date)
+    where('start_date <= ? and end_date >= ?', date, date).select do |service|
+      service.weekdays[date.wday]
+    end - removed_on(date) + added_on(date)
```

Comment:
You can do this sort of sorcery on line 27?

Reply:
Sure, array addition and subtraction.